r1cs/gadgets/std/src/fields/fp.rs

```diff
@@ -489,20 +489,34 @@ impl<F: PrimeField> ToBitsGadget<F> for FpGadget<F> {
     }
 }
 
-// Pack a slice of Boolean gadgets into one or several field elements, bundling at most
-// `F::Params::CAPACITY` many in a single field element (to allow efficient unpacking).
+// Pack a slice of Boolean gadgets into a field element, bundling at most
+// `F::Params::CAPACITY` many in the constructed field element (to allow efficient unpacking);
+// the remaining Boolean gadgets found in the slice are ignored.
 // The bundling regards the Booleans in big endian order.
 impl<F: PrimeField> FromBitsGadget<F> for FpGadget<F> {
     fn from_bits<CS: ConstraintSystemAbstract<F>>(
         mut cs: CS,
         bits: &[Boolean],
     ) -> Result<Self, SynthesisError> {
-        // We can safely pack up to CAPACITY bits
-        let bits = bits.chunks(F::Params::CAPACITY as usize).next().unwrap();
+
+        let mut bits_chunk = bits.chunks(F::Params::CAPACITY as usize);
+
+        // Pre checks
+        if bits_chunk.len() == 0 {
+            return Err(SynthesisError::Other("empy slice of Booleans provided".to_string()));
+        }
+
+        if bits_chunk.len() > 1 {
+            return Err(
+                SynthesisError::Other("You can reconstruct only 1 field element by calling this function: bits cannot be bigger than Field::CAPACITY".to_string())
+            );
+        }
```

Comment:
I think that this code can be safely replaced by a simple check on `bits.len()` (which should be `> 0` and `<= F::Params::CAPACITY`)

Reply:
Actually I think that with this change, `bits` could also be the empty slice. In this case the function would return the zero element (which I find reasonable)